### ISSUE TITLE
Proposal: Modify, a slightly modified version of Merge

### ISSUE THREAD

**Issue by Evertt** [sindresorhus/type-fest#604]
`Modify` is like `Merge`, but it allows you to remove properties from the resulting type if you define that property in the `Source` with type `never`.

See this playground, it will explain everything:
https://tsplay.dev/m0zRrw

**Comment by sindresorhus** [comment]
What's your argument for making a new type instead of supporting this in `Merge`?

**Comment by Evertt** [comment]
Yeah no good reason, I'd be very happy if this was just supported in `Merge` by default. And even better if it could be supported in `MergeDeep` as well.
